components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2333,6 +2351,26 @@ private String doUserAuthorization(OAuthMessage oAuthMessage, String sessionData
 
         OAuth2Parameters oauth2Params = getOauth2Params(oAuthMessage);
         AuthenticatedUser authenticatedUser = getLoggedInUser(oAuthMessage);
+
+        // Here we validate all scopes, response type, client before consent page
+        HttpRequestHeaderHandler httpRequestHeaderHandler = new HttpRequestHeaderHandler(oAuthMessage.getRequest());
+        OAuth2AuthorizeReqDTO authzReqDTO =
+                buildAuthRequest(oauth2Params, oAuthMessage.getSessionDataCacheEntry(), httpRequestHeaderHandler);
+        try {
+            handleAuthorizationBeforeConsent(oAuthMessage, oauth2Params, authzReqDTO);
+        } catch (IdentityOAuth2ScopeValidationException e) {
+            LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, null,
+                    OAuthConstants.LogConstants.FAILED, "Error occurred when processing the authorization"
```

Comment:
@chamathns Can we check whether this follow required best practices related to diagnostic logs?

Reply:
The result message here seems too generic. With diagnostic logs, the user should be able to troubleshoot and fix the issues. Hence let's include more context information (application, tenant, user Id, ...) as params for diagnostic logs and improve the result message.